### PR TITLE
feat(attributes-panel): add colorize by attribute section

### PR DESCRIPTION
**Pull request by kirillovzah**

### CODE REVIEW

File: src/components/attributes-panel/attributes-stats/color-value-item.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const ValueItemArrow = styled(DropdownUp)<{ deg: number; visible: boolean }>`
+  width: 15px;
+  height: 17px;
+  fill: ${({ theme }) => theme.colors.fontColor};
+  transform: ${({ deg }) => `rotate(${deg}deg)`};
+`;
+
+type ColorValueItemProps = {
+  deg?: number;
+  yearCount?: number;
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/color-value-item.tsx:37]: year?

File: src/components/attributes-panel/attributes-stats/attribute-stats.tsx [hunk 2]
```diff
@@ -295,15 +323,41 @@ export const AttributeStats = ({
...
+                  </FadeContainer>
+                  <ColorizeValuesList>
+                    <ColorValueItem
+                      arrowVisibility={VISIBLE}
+                      deg={-90}
+                      yearCount={statistics.min}
+                    />
+                    <ColorValueItem
+                      arrowVisibility={NOT_VISIBLE}
+                      yearCount={Math.floor(statistics.avg)}
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/attribute-stats.tsx:349]: I won't be able to visualize it with distribution avg calculated. Let's calculate it from min and max values

File: src/components/attributes-panel/attributes-stats/attribute-stats.tsx [hunk 3]
```diff
@@ -295,15 +323,41 @@ export const AttributeStats = ({
...
-
-          <AttributeColorize>
-            <ColorizeTitle>{COLORIZE_BY_ATTRIBUTE}</ColorizeTitle>
-            <ToggleSwitch
-              id={"colorize-by-attribute"}
-              checked={true}
-              onChange={onColorizeByAttributeClick}
-            />
-          </AttributeColorize>
+          {statistics?.max && statistics?.avg && (
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/attribute-stats.tsx:326]:
```suggestion
          {statistics?.max && statistics?.min && (
```

File: src/components/attributes-panel/attributes-stats/attribute-stats.tsx [hunk 4]
```diff
@@ -120,11 +123,34 @@ const SpinnerContainer = styled.div<VisibilityProps>`
...
+  margin-bottom: 36px;
+`;
+
 const HISTOGRAM = "histogram";
 const MOST_FREQUENT_VALUES = "mostFrequentValues";
 const COLORIZE_BY_ATTRIBUTE = "Colorize by Attribute";
 const VALUE_TITLE = "Value";
 const COUNT_TITLE = "Count";
+const VISIBLE = true;
+const NOT_VISIBLE = false;
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/attribute-stats.tsx:153]: Don't use these constants. They are excess

File: src/components/attributes-panel/attributes-panel.tsx [hunk 5]
```diff
@@ -141,6 +141,8 @@ export const AttributesPanel = ({
   const [selectedAttributeStatsInfo, setSelectedAttributeStatsInfo] =
     useState<StatisticsInfo | null>(null);
   const [selectedAttributeName, setSelectedAttributeName] = useState("");
+  const [showColorizeByAttribute, setShowColorizeByAttribute] =
```
Comment (belom88) [inline src/components/attributes-panel/attributes-panel.tsx:144]: Move this state to `AttributeStats` level

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/components/attributes-panel/attributes-panel.tsx [hunk 6]
```diff
@@ -141,6 +141,7 @@ export const AttributesPanel = ({
   const [selectedAttributeStatsInfo, setSelectedAttributeStatsInfo] =
     useState<StatisticsInfo | null>(null);
   const [selectedAttributeName, setSelectedAttributeName] = useState("");
+
```
Comment (belom88) [inline src/components/attributes-panel/attributes-panel.tsx:144]: too many extra lines

File: src/components/attributes-panel/attributes-stats/attribute-stats.spec.tsx [hunk 7]
```diff
@@ -81,8 +81,7 @@ beforeAll(() => {
 
 describe("AttributeStats", () => {
   it("Should render Attribute Stats", async () => {
-    const onColorizeByAttributeClick = jest.fn();
-
+
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/attribute-stats.spec.tsx:84]: too many extra lines

File: src/components/attributes-panel/attributes-stats/attribute-stats.spec.tsx [hunk 8]
```diff
@@ -115,22 +113,23 @@ describe("AttributeStats", () => {
       }
     }
 
+    const histogramIcon = screen.getByText("Histogram").firstElementChild
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/attribute-stats.spec.tsx:116]:
```suggestion
    const histogramIcon = screen.getByText("Histogram").firstElementChild;
```

Comment (belom88) [reply src/components/attributes-panel/attributes-stats/attribute-stats.spec.tsx:116]: Declare the variable in the place it is used

File: src/components/attributes-panel/attributes-stats/attribute-stats.tsx [hunk 9]
```diff
@@ -277,33 +296,63 @@ export const AttributeStats = ({
               <HistograpPanel>
                 <HistogramTitle>
                   Histogram
-                  <HistogamArrow
-                    data-testid="histogram-arrow"
-                    open={showHistogram}
-                    onClick={() => setShowHistogram((prevValue) => !prevValue)}
+                  <ExpandIcon
+                    data-testid={"histogram-arrow"}
```
Comment (belom88) [inline src/components/attributes-panel/attributes-stats/attribute-stats.tsx:300]: Can we remove `data-testid` yet?

File: src/utils/calculate-average-value.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export const calculateAverageValue = (
+    minValue: number,
+    maxValue: number
+  ): number => {
+    const valueSum = minValue + maxValue;
+
+    return Math.floor(valueSum / 2);
+  };
```
Comment (belom88) [inline src/utils/calculate-average-value.ts:8]: New line in the end needed

File: src/utils/hooks/use-expand.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import { useCallback, useState } from "react";
```
Comment (belom88) [inline src/utils/hooks/use-expand.ts:1]: Add tests

File: src/utils/calculate-average-value.spec.ts [hunk 12]
```diff
@@ -0,0 +1,9 @@
+import { calculateAverageValue } from './calculate-average-value';
+
+describe("Format Memory", () => {
```
Comment (belom88) [inline src/utils/calculate-average-value.spec.ts:3]:
```suggestion
describe("Calculate average value", () => {
```

**Review by belom88 — CHANGES_REQUESTED** [review]

**Review by belom88 — CHANGES_REQUESTED** [review]
Just "0" in the place
image

**Comment by ibgreen** [comment]
Nit: Way too many decimals. Suggest truncating to 3 or possibly 4 significant digits.

File: src/utils/calculate-average-value.ts [hunk 13]
```diff
@@ -5,4 +5,5 @@ export const calculateAverageValue = (
     const valueSum = minValue + maxValue;
 
     return Math.floor(valueSum / 2);
-  };
\ No newline at end of file
+  };
+
```
Comment (belom88) [inline src/utils/calculate-average-value.ts:9]: Something wrong here. Probably - 9nth line is not empty

**Review by belom88 — APPROVED** [review]

**Pull request merged**